agent/config/config.go

```diff
@@ -195,8 +195,16 @@ func (cfg *Config) Merge(rhs Config) *Config {
 
 	for i := 0; i < left.NumField(); i++ {
 		leftField := left.Field(i)
-		if utils.ZeroOrNil(leftField.Interface()) {
-			leftField.Set(reflect.ValueOf(right.Field(i).Interface()))
+		switch leftField.Interface().(type) {
+		case BooleanDefaultFalse, BooleanDefaultTrue:
+			str, _ := json.Marshal(reflect.ValueOf(leftField.Interface()).Interface())
```

Comment:
im open to explicitly checking NotSet value here but json logic already handles it so kept it this way.